Test_dialogue_overlay.py

```diff
@@ -0,0 +1,246 @@
+import unittest
```

Comment:
can we have a directory for tests?

Reply:
also move unit_test_data into the test directory

Reply:
Done